### PR TITLE
Some (mostly coloring-focused) bug fixes

### LINKED ISSUE #760: "Incomplete" interpolation for sequential / diverging color maps
When selecting a sequential / diverging color map (e.g. Viridis), the interpolation is incomplete: the "end" color isn't the end of the color map. See the example below, where the end color should be #fee825 (yellow) but is actually #8cb373 (blue-ish green).

image

This notably doesn't seem to be a problem when using the "continuous values" scaling:

image

I suspect that [this line](https://github.com/biocore/emperor/blob/023b6ecb761c31cd7f60a2e38e418d71199eb4e1/emperor/support_files/js/color-view-controller.js#L468) in `color-view-controller.js` may be causing these problems, but I'm not sure.

### LINKED ISSUE #762: Update chroma.js version
As mentioned in https://github.com/biocore/empress/pull/159#issuecomment-625580999, the version used by Empress and Emperor is a few years out of date, so this results in tiny precision problems.

### PR DESCRIPTION
**Pull request by fedarko**
Closes #760 and closes #762. Also fixes that `naturalSort()` Infinity issue we talked about (there isn't an open issue for that though).

The JavaScript tests have ~~mostly~~ all been updated. This involved updating both the expected colors from the interpolation fixes, as well as updating slightly different colors due to the new Chroma.js version. (For the two tests that directly checked gradient SVG equality, I elected to just copy in the new expected SVG over the old expected SVG markup -- changing each hex color manually seemed kinda daunting, as shown in the QUnit diff below.)

errors

I also tested this manually, and can confirm that this resolves #760 -- see below, where the interpolated Viridis colors now extend to cover the full color map.

image

### CODE REVIEW

File: emperor/support_files/js/util.js [hunk 1]
```diff
@@ -13,8 +13,12 @@ define(['underscore'], function(_) {
     var numericPart = [], alphaPart = [], result = [];
 
     // separate the numeric and the alpha elements of the array
+    // Note that NaN and +/- Infinity are not considered numeric elements
     for (var index = 0; index < list.length; index++) {
-      if (isNaN(parseFloat(list[index]))) {
+      if (
+          isNaN(parseFloat(list[index])) ||
```
Comment (wasade) [inline emperor/support_files/js/util.js:19]: Worth doing the parse to float a single time?

File: tests/javascript_tests/test_util.js [hunk 2]
```diff
@@ -118,6 +118,47 @@ requirejs(['jquery', 'underscore', 'util'], function($, _, util) {
...
+     *
+     * Test that strings like "Infinity" are sorted with words in
+     * naturalSort's output, rather than with numbers. This test was adapted
+     * from Empress.
+     *
+     */
+    test("Test that naturalSort doesn't treat Infinity / NaN as numbers", function () {
+      var eles = [
+        "1",
+        "2",
```
Comment (wasade) [inline tests/javascript_tests/test_util.js:131]: is it important to include things like 1e10?

File: emperor/support_files/js/util.js (hunk 1 above)
Comment (fedarko) [reply emperor/support_files/js/util.js:19]: yeah, would be nice (and should make this if statement fit on one line i guess). will resolve tomorrow

File: tests/javascript_tests/test_util.js (hunk 2 above)
Comment (fedarko) [reply tests/javascript_tests/test_util.js:131]: I hadn't thought of that! Would be nice to have tested. That being said, it looks like in JS scientific notation stuff like 1e10 is interpreted correctly as a number by `parseFloat()`, so I think this should be handled as expected already (but ofc would be best to explicitly test for that :)

```js
> parseFloat("1e10");
10000000000
```

Comment (wasade) [reply tests/javascript_tests/test_util.js:131]: Excellent :)

**Comment by fedarko** [comment]
Comments have been addressed. It's worth noting that scientific notation numbers like `1e2` are handled as expected by `naturalSort()` (and this is now explicitly tested for), _but_ the legend is still not sorting things correctly since this PR doesn't address #761. So datasets can still look something like

image

... where the colors are assigned correctly, but the legend is out of order.

If y'all would like me to I can try to address #761 in this PR as well, but I'm not sure where to look for that.

Thanks!

**Pull request merged**